### PR TITLE
[AB#39261] Added video cue point support to Media and Catalog services

### PR DESCRIPTION
**Pull request by IrinaBazyleva**
# Pull Request

## Prerequisites

- [x] My code follows the coding conventions
- [x] All tests pass

## Description

Added video cue point support to Media and Catalog services.
Media service on start registers 3 video cue point types:
- Intro out
- Chapter marker
- Outro in

When a media asset video is published, the cue points associated with it are also published as part of the video metadata.

Catalog API endpoints now include video cue points information.

## Migration

The service account associated with the Media Service should have assigned Encoding service permission `Cue Point Types: Declare`. This can be achieved by running locally command `yarn setup` from the Media Service solution root, or by adjusting the service account permissions in the Admin Portal.

### CODE REVIEW

**Comment by IrinaBazyleva** [comment]
Please note that pull request [[AB#38416] Added support for Channels to Catalog Service](https://github.com/Axinom/mosaic-media-template/pull/37) includes new database migrations. If this pull request is merged into dev before another pull request that also modifies the database, it may be necessary to regenerate the migrations for the second pull request.

File: services/media/service/scripts/setup.ts [hunk 1]
```diff
@@ -46,6 +46,10 @@ async function main(): Promise<void> {
         serviceId: 'ax-image-service',
         permissions: ['IMAGE_TYPES_DECLARE'],
       },
+      {
```
Comment (AxTrusov) [inline services/media/service/scripts/setup.ts:49]: I'd mention in the PR description that `yarn setup` should be re-run in development and in deployed environments configured Service Account should be adjusted to include an extra permission, as a migration step to enable this functionality.

File: services/media/service/src/domains/common/register-messaging.ts [hunk 2]
```diff
@@ -25,6 +28,10 @@ export const registerCommonMessaging = (
       ImageServiceMultiTenantMessagingSettings.DeclareImageTypes,
       config,
     ).sendCommand(),
+    new RascalConfigBuilder(
```
Comment (AxTrusov) [inline services/media/service/src/domains/common/register-messaging.ts]: Can you also add handlers and listen for Success/Fail events? Just logging the received payload with a fitting message is enough. Failed - ERROR log level, Success - INFO log level.

**Review by AxTrusov — APPROVED** [review]
Looks good. Can be merged after the comments are resolved.

File: services/media/service/scripts/setup.ts (hunk 1 above)
Comment (IrinaBazyleva) [reply services/media/service/scripts/setup.ts:49]: Pull request description was updated to contain migration steps.

File: services/media/service/src/domains/common/register-messaging.ts (hunk 2 above)
Comment (IrinaBazyleva) [reply services/media/service/src/domains/common/register-messaging.ts]: Added missing RabbitMQ handlers.

**Pull request merged**